aten/src/ATen/native/cuda/TensorTransformations.cu

```diff
@@ -0,0 +1,121 @@
+#include "ATen/native/TensorTransformations.h"
+
+#include "ATen/cuda/detail/IndexUtils.cuh"
+#include "ATen/NativeFunctions.h"
+#include "ATen/cuda/CUDATensorMethods.cuh"
+#include "ATen/cuda/CUDATypeConversion.cuh"
+
+namespace at {
+namespace native {
+
+#define AT_APPLY_THREADS_PER_BLOCK 32 * 16
+#define AT_APPLY_BLOCKS_PER_SM 4
+
+template <typename scalar_t, typename IndexType>
+#if __CUDA_ARCH__ >= 350
+__launch_bounds__(AT_APPLY_THREADS_PER_BLOCK, AT_APPLY_BLOCKS_PER_SM)
+#endif
+__global__ void
+kernel_pointwise_flip_apply2(const cuda::detail::TensorInfo<scalar_t, IndexType> in_tensor_info,
+                          cuda::detail::TensorInfo<scalar_t, IndexType> out_tensor_info,
+                          IndexType N,
+                          int flip_dim,
+                          int64_t total_dims) {
+  for (IndexType linear_index = blockIdx.x * blockDim.x + threadIdx.x; linear_index < N; linear_index += gridDim.x * blockDim.x) {
+    int64_t cur_indices = linear_index, rem = 0, dst_offset = 0;
+    for (int64_t i = 0; i < total_dims; i++) {
```

Comment:
You know that total_dims here is 2, right? You've collapse the dims. If you make it a compile-time constant, loop will be unrolled and be more efficient.

Reply:
Ah, great point! I will remove the for loop.

Reply:
Removing for loop gives way better performance!

```
data_cuda = torch.arange(1000000, device=cuda).view(1000,1000)
%timeit data_cuda.flip(0)
--------------------
10000 loops, best of 3: 90 µs per loop
```